tests/test_vrf.py

```diff
@@ -0,0 +1,1395 @@
+
+            # remove ip addr, intf, vrf, vlan member, vlan cfgs
+            for cfg_name, attrs in reversed(cfg_attrs_map.items()):
+                src_template = 'vrf/vrf_capacity_{}_cfg.j2'.format(cfg_name)
+                render_file = '/tmp/vrf_capacity_del_{}_cfg.json'.format(cfg_name)
+                duthost.template(src=src_template, dest=render_file)
+                duthost.shell("sonic-cfggen -j {} --write-to-db".format(render_file))
+
+                time.sleep(attrs['remove_sleep_time'])
+
+        duthost.shell("logger -p INFO -- '-------- {} end!!! ---------'".format(request.cls.__name__))
+
+    def test_ping(self, duthost, random_vrf_list):
+        dut_extra_vars = {
+            'vrf_name_tpl':     self.vrf_name_tpl,
+            'random_vrf_list':  random_vrf_list,
+            'ip1':              self.ipnet1,
+            'ip2':              self.ipnet2
+        }
+        duthost.host.options['variable_manager'].extra_vars = dut_extra_vars
+        duthost.template(src='vrf/vrf_capacity_ping.j2', dest='/tmp/vrf_capacity_ping.sh', mode="0755")
+
+        duthost.shell('/tmp/vrf_capacity_ping.sh')
+
+    def test_ip_fwd(self, duthost, ptfhost, host_facts, testbed, random_vrf_list):
+        ptf_port1 = g_vars['vrf_intf_member_port_indices']['Vrf1']['Vlan1000'][1]
+        ptf_port2 = g_vars['vrf_intf_member_port_indices']['Vrf2']['Vlan2000'][1]
+        dst_ips = [str(IPNetwork(self.route_prefix)[1])]
+
+        ptf_runner(ptfhost,
+                    "ptftests",
+                    "vrf_test.CapTest",
+                    platform_dir="ptftests",
+                    params={
+                            "testbed_type": testbed['topo'],
+                            "router_mac": host_facts['ansible_Ethernet0']['macaddress'],
+                            "src_ports": [ptf_port1],
+                            "dst_ports": [[ptf_port2]],
+                            "dst_ips":  json.dumps(dst_ips).replace('"', r'\"'),
+                            "random_vrf_list": random_vrf_list,
+                            "base_vid": self.base_vid
+                            },
+                    log_file="/tmp/vrf_capacity_test.CapTest.log")
+
+
+class TestVrfUnbindIntf():
+    c_vars = {}
+
+    @pytest.fixture(scope="class", autouse=True)
+    def setup_vrf_unbindintf(self, duthost, ptfhost, testbed, cfg_facts):
+        # -------- Setup ----------
+        duthost.shell("config interface vrf unbind PortChannel0001")
+
+        # wait for neigh/route flush
+        time.sleep(5)
+
+        # -------- Testing ----------
+        yield
+
+        # -------- Teardown ----------
+        duthost.shell("config interface vrf bind PortChannel0001 Vrf1")
+        for ver, ips in g_vars['vrf_intfs']['Vrf1']['PortChannel0001'].iteritems():
+            for ip in ips:
+                duthost.shell("config interface ip add PortChannel0001 {}".format(ip))
+        time.sleep(10)  # wait for bgp session re-established.
```

Comment:
I think should check the BGP session status here, since it is critical to the below test.

Reply:
Agreed. Check bgp session status after rebind intf(in `TestVrfUnbindIntf`) and restore vrf(in `TestVrfDeletion`). In addition, neigh/fib tests are added after rebind/restore.